django/utils/formats.py

```diff
@@ -172,18 +172,19 @@ def localize(value, use_l10n=None):
     If use_l10n is provided and is not None, that will force the value to
     be localized (or not), overriding the value of settings.USE_L10N.
     """
+    if isinstance(value, six.string_types):
+        return value
     if isinstance(value, bool):
         return mark_safe(six.text_type(value))
-    elif isinstance(value, (decimal.Decimal, float) + six.integer_types):
```

Comment:
I'd keep it as-is then. It's confusing to combine cosmetic cleanups like that with optimizations anyway.

argument I read: http://stackoverflow.com/a/22783232/5112